### DISCUSSION TITLE
"Lazy" tables when converting from HTML to GitHub Flavored Markdown

### DISCUSSION THREAD

**Discussion by johnmapeson** [jgm/pandoc#9691] [Q&A]
Is it possible to ask Pandoc to use "lazy" syntax for tables when converting from HTML to GitHub Flavored Markdown?

That is, instead of converting

```
<table>
  <tr>
    <td>one</td>
    <td>two</td>
  </tr>
  <tr>
    <td>a very long sentence</td>
    <td>a very very long sentence</td>
  </tr>
<table>
```

to

```
| one                  | two                       |
| -------------------- | ------------------------- |
| a very long sentence | a very very long sentence |
```

I would prefer to convert it to

```
| one | two |
| - | - |
| a very long sentence | a very very long sentence |
```

The command I use currently is:

```
pandoc from.htm -t gfm-raw_html -o to.md --wrap=none
```

**Comment by jgm** [comment]
I'm not sure why you'd prefer that ugly table, which is much harder to read in the source.  But no, I'm not inclined to provide an option for that. The markdown writer has to make a number of decisions about formatting, and making them all configurable would lead to a combinatorial explosion of complexity.

**Reply by johnmapeson** [reply]
> I'm not sure why you'd prefer that ugly table, which is much harder to read in the source.

I agree this style is ugly. But here is a real-table from one article on the internet:

<img width="653" alt="image" src="https://github.com/jgm/pandoc/assets/129062294/ffa0bfb9-eb47-43ac-9a49-710329d86609">

After converting to GFM it looks as follows:

```
| **Method**    | **What is it?**                                                                                                                                      | **Example**                                                                                                                                                      |
|---------------|------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Time Blocking | Dividing the day into blocks of time with each block dedicated to accomplishing a specific task or activity and only that specific task or activity. | "I will write every day from 9am to 11am."                                                                                                                       |
| Task Batching |                                                                                                                                                      | "I will answer all of my emails at 3pm."                                                                                                                         |
| Day Theming   |                                                                                                                                                      | "Every Monday, I will focus on content creation. Every Tuesday, I will focus on content promotion. Every Wednesday, I will focus on research and ideation. Etc." |
| Time Boxing   |                                                                                                                                                      | "I will write 1,000 words between 9am and 11am tomorrow."    
```

This looks fine (well, the rows are extremely long, but at least the columns are nicely aligned), but only if you have line wrapping turned off. If you have line wrapping turned on, and the window is, for example, about 100 columns width, such a table will look as follows,

<img width="688" alt="image" src="https://github.com/jgm/pandoc/assets/129062294/afa61763-022b-49b9-a8df-56ac80256da0">

which is, I would say, is much uglier than

<img width="678" alt="image" src="https://github.com/jgm/pandoc/assets/129062294/1b8ca842-bab6-4684-9c93-4eee6fca6a82">

**Reply by jgm** [reply]
I see.  Well, maybe we could have the writer omit the alignment if the table width is greater than COLUMNS.

**Reply by johnmapeson** [reply]
This seems to be a great idea.

**Reply by jgm** [reply]
Go and ahead and submit an issue requesting this.

**Reply by johnmapeson** [reply]
Here it is: https://github.com/jgm/pandoc/issues/9696.
